scripts/src/scrubMessages.ts

```diff
@@ -61,7 +67,26 @@ const scrubMessages = async ({ db }: { db: Db }) => {
         embedding: "$messages.embedding",
         rating: "$messages.rating",
         references: "$messages.references",
-      } satisfies Partial<Record<keyof ScrubbedMessage, string | number>>,
+        doNotAnswer: "$messages.doNotAnswer",
+      } satisfies Record<
+        Exclude<
+          // This protects against unknown entries in the $project stage and
+          // ensures all of the fields that we do want are projected. We can't
+          // just use ScrubbedMessage/SomeMessage because we want the union of
+          // all possible Message-type keys.
+          | keyof UserMessage
+          | keyof AssistantMessage
+          | keyof SystemMessage
+          | keyof ScrubbedMessage,
+          // Add keys to omit from the projection below.
+          | "id"
+          | "content"
+          | "preprocessedContent"
```

Comment:
Are these not already included? I figured since the all the `FooMessage` types extend `Message` these would already be in the union.

Reply:
We're explicitly trying to exclude these fields here so they don't get projected into the scrubbed version, if that makes sense.

Reply:
Of course they won't get projected as-is, but by adding it to the `requires` clause we encode the intention to keep them out.